File: Daily_Coding_challenge/557-done/557.cpp
#include<bits/stdc++.h>

using namespace std;

int count_x_in_square_matrix_o_n(int n,int x){
    int count=0;
    if (n ==0 || x ==0 || x > n*n) 
        return 0;
    for(int i=1;i<=n;i++) {
        if ((x%i==0) && (x/i <=n)){
            count+=1;
            //cout <<i <<endl;
        }
    }
    return count;
}



int count_x_in_square_matrix_o_n_2(int n,int x){
    int count=0, i=1;
    if (n ==0 || x ==0 || x > n*n) 
        return 0;
    
    while (i*i <x) {
        if ((x%i==0) && (x/i <=n))
            count+=2;
        //cout <<i <<" " << count<<endl;
        i+=1;
        if (i*i ==x)
            count+=1;
    }
    return count;
}

int main()
{
    cout <<count_x_in_square_matrix_o_n_2(6,12)<<endl;
    return 0;
}

File: Daily_Coding_challenge/557-done/557.py
def count_x_in_square_matrix_o_n(n,x):  
    cnt=0;
    for i in range(2,n+1):
        if x % i == 0 and x/i<=n:
            cnt+=1
    return cnt

def count_x_in_square_matrix_o_n_2(n,x):
    if n  ==0 and x ==0 and x > n*n:
        return 0
    if n == 1:
        if x !=1:
            return 0;
        return 1;
    i =1    
    cnt=0;
    while (i*i <x) :
        if x %i ==0 and x <= n *i:
            cnt+=2
        i+=1
        if i*i == x :
            cnt+=1
            
        #print("{} {} ".format(i,cnt))

    return cnt
if __name__ == "__main__":
    print(count_x_in_square_matrix_o_n_2(6,14))
    
    
    """
    count_x_in_square_matrix_o_n
        time complexity =  O(n)
        space complexity = O(1)

    count_x_in_square_matrix_o_n_2
        time complexity =  O(sqrt(n))
        space complexity = O(1)
    """